consoles/vnc_base.pm

```diff
@@ -178,26 +171,15 @@ sub mouse_hide ($self, $args) {
 
 sub mouse_set ($self, $args) {
     die "Need x/y arguments" unless (defined $args->{x} && defined $args->{y});
-
-    # TODO: for framebuffers larger than 1024x768, we need to upscale
     $self->_mouse_move(int($args->{x}), int($args->{y}));
     return {};
 }
 
 sub mouse_button ($self, $args) {
     my $button = $args->{button};
     my $bstate = $args->{bstate};
-
-    my $mask = 0;
-    if ($button eq 'left') {
-        $mask = $bstate;
-    }
-    elsif ($button eq 'right') {
-        $mask = $bstate << 2;
-    }
-    elsif ($button eq 'middle') {
-        $mask = $bstate << 1;
-    }
+    my %mask = (left => $bstate, right => $bstate << 2, middle => $bstate << 1);
+    my $mask = $mask{button} // 0;
```

Comment:
`$mask{button}` misses a `$`.
```suggestion
    my $mask = {left => $bstate, right => $bstate << 2, middle => $bstate << 1}->{$button} // 0;
```

Reply:
done, interesting syntax but makes sense and looks nice :)